### PR TITLE
Feature/katex2

### PR DESCRIPTION
**Pull request by Entkenntnis**
adding support for math formulas

### CODE REVIEW

**Comment by Entkenntnis** [comment]
I've created a new branch because there are some conflicts while merging

File: packages/plugin-text/src/plugins/katex/index.tsx [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+
+  const handleChange = debounce((formula: string) => {
+    editor.setNodeByKey(node.key, {
+      type: node.type,
+      data: {
+        formula,
+        inline: node.data.get('inline')
+      }
+    })
+  }, 50)
```
Comment (Knorrke) [inline packages/plugin-text/src/plugins/katex/index.tsx]: This looks pretty low. Maybe you don't need a debounce at all?

File: packages/plugin-text/src/plugins/katex/index.tsx [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+            }}
+          />
+          <span> Preview: </span>
+          <Math
+            formula={node.data.get('formula')}
+            inline={node.data.get('inline')}
+          />
+        </Overlay>
+      ) : null}
+      {graphicalEditor ? <div>GUPPY EDITOR</div> : null}
```
Comment (Knorrke) [inline packages/plugin-text/src/plugins/katex/index.tsx]: remove this :-)

File: packages/plugin-text/src/plugins/katex/math.component.tsx [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+      return null
+    }
+
+    const Component = inline ? InlineMath : BlockMath
+
+    return (
+      <Component
+        math={formula}
+        renderError={error =>
+          handleError(formula, error, !inline, oldErrorPosition as number)
```
Comment (Knorrke) [inline packages/plugin-text/src/plugins/katex/math.component.tsx]: you shouldn't have to explicitly state `as number` here. This is an indication that there is something wrong with the types.

File: packages/plugin-text/src/plugins/katex/math.component.tsx [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+        }
+      />
+    )
+  }
+}
+
+export interface MathProps {
+  formula?: string
+  inline?: boolean
+  oldErrorPosition?: unknown
```
Comment (Knorrke) [inline packages/plugin-text/src/plugins/katex/math.component.tsx]: why unknown and not number?

File: packages/ui/src/elements/settings-overlay.tsx [hunk 5]
```diff
@@ -235,7 +235,7 @@ export const HoveringOverlay: React.FunctionComponent<{
   )
 }
 
-const ConfigIconContainer = styled.div({
+const ConfigIconContainer = styled.span({
```
Comment (Knorrke) [inline packages/ui/src/elements/settings-overlay.tsx]: revert this please :-)

File: yarn.lock [hunk 6]
```diff
@@ -10649,6 +10682,16 @@ react-transition-group@^2.2.1:
     prop-types "^15.6.2"
     react-lifecycles-compat "^3.0.4"
 
+react@^16.5.2:
```
Comment (Knorrke) [inline yarn.lock]: ping @inyono is this a problem? Like: Two React versions? Is required by `react-mathquill`

File: packages/plugin-text/src/plugins/katex/index.tsx (hunk 2 above)
Comment (Entkenntnis) [reply packages/plugin-text/src/plugins/katex/index.tsx]: done

File: packages/plugin-text/src/plugins/katex/index.tsx (hunk 1 above)
Comment (Entkenntnis) [reply packages/plugin-text/src/plugins/katex/index.tsx]: debounce is removed

File: packages/ui/src/elements/settings-overlay.tsx (hunk 5 above)
Comment (Entkenntnis) [reply packages/ui/src/elements/settings-overlay.tsx]: revert

File: packages/plugin-text/src/plugins/katex/math.component.tsx (hunk 4 above)
Comment (Entkenntnis) [reply packages/plugin-text/src/plugins/katex/math.component.tsx]: changed to number

File: packages/plugin-text/src/plugins/katex/math.component.tsx (hunk 3 above)
Comment (Entkenntnis) [reply packages/plugin-text/src/plugins/katex/math.component.tsx]: added if-statement instead

**Pull request merged**